projectroles/utils.py

```diff
@@ -95,3 +109,197 @@ def get_app_names():
             else:
                 ret.append(s[0])
     return sorted(ret)
+
+
+class SidebarContent:
```

Comment:
I think we need to unify the sidebar and user dropdown helpers into a single class, see my comment further on. Perhaps the naming could be something like `AppLinkContent`.